src/sql/platform/connection/common/connectionManagementService.ts

```diff
@@ -411,110 +388,90 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 				showFirewallRuleOnError: true
 			};
 		}
-		return new Promise<IConnectionResult>(async (resolve, reject) => {
-			if (callbacks.onConnectStart) {
-				callbacks.onConnectStart();
-			}
-			let tokenFillSuccess = await this.fillInOrClearAzureToken(connection);
-			if (!tokenFillSuccess) {
-				throw new Error(nls.localize('connection.noAzureAccount', "Failed to get Azure account token for connection"));
-			}
-			this.createNewConnection(uri, connection).then(connectionResult => {
-				if (connectionResult && connectionResult.connected) {
-					// The connected succeeded so add it to our active connections now, optionally adding it to the MRU based on
-					// the options.saveTheConnection setting
-					let connectionMgmtInfo = this._connectionStatusManager.findConnection(uri);
-					this.tryAddActiveConnection(connectionMgmtInfo, connection, options.saveTheConnection);
-
-					if (callbacks.onConnectSuccess) {
-						callbacks.onConnectSuccess(options.params, connectionResult.connectionProfile);
-					}
-					if (options.saveTheConnection) {
-						this.saveToSettings(uri, connection).then(value => {
-							this._onAddConnectionProfile.fire(connection);
-							this.doActionsAfterConnectionComplete(value, options);
-						});
-					} else {
-						connection.saveProfile = false;
-						this.doActionsAfterConnectionComplete(uri, options);
-					}
-					if (connection.savePassword) {
-						this._connectionStore.savePassword(connection).then(() => {
-							resolve(connectionResult);
-						});
-					} else {
-						resolve(connectionResult);
-					}
-				} else if (connectionResult && connectionResult.errorMessage) {
-					this.handleConnectionError(connection, uri, options, callbacks, connectionResult).then(result => {
-						resolve(result);
-					}).catch(handleConnectionError => {
-						if (callbacks.onConnectReject) {
-							callbacks.onConnectReject(handleConnectionError);
-						}
-						reject(handleConnectionError);
+		if (callbacks.onConnectStart) {
+			callbacks.onConnectStart();
```

Comment:
You are slightly changing the behavior here (and other places) because now there's some parts of this that are done outside of the async part. So before while this call to onConnectStart would be called inside the promise now it's not so a long-running callback could potentially affect the performance of this method.